src/main/java/org/jahia/modules/appshell/Main.java

```diff
@@ -144,49 +218,119 @@ private void setCustomAttributes(JahiaUser currentUser, HttpServletRequestWrappe
         wrapper.setAttribute("config", config.toString());
     }
 
+    public void updateBundleInfos() {
+        List<Bundle> packages = getPackages();
+
+        List<Pattern> newUrlPatterns = new ArrayList<>();
+        Map<Bundle, JSONObject> newBundleDescriptors = new LinkedHashMap<>();
+
+        for (Bundle bundle : packages) {
+            JSONObject pkgJson = null;
+            try {
+                pkgJson = new JSONObject(IOUtils.toString(bundle.getResource(PACKAGE_JSON) != null ? bundle.getResource(PACKAGE_JSON) : bundle.getResource(JAHIA_JSON)));
+                if (pkgJson.has(JAHIA) && pkgJson.getJSONObject(JAHIA).has(URLPATTERNS)) {
+                    Object urlPatternsObj = pkgJson.getJSONObject(JAHIA).get(URLPATTERNS);
+                    List<String> urlPatternArray = getStringList(urlPatternsObj);
+                    for (String urlPattern : urlPatternArray) {
+                        Pattern compiledUrlPattern = Pattern.compile(urlPattern);
+                        newUrlPatterns.add(compiledUrlPattern);
+                    }
+
```

Comment:
Yes agreed. Again here I was trying to fix the existing code rather than refactor it too much, but I guess it does make sense to clean it at the same time.